### PR TITLE
doc: Update container package overview

### PR DESCRIPTION
**Pull request by JonasBakys0**
**Description:**

- Updates incorrect overview for container package
- [issue](https://github.com/gofr-dev/gofr/issues/991)

**Checklist:**

-   [ ] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [ ] All new code is covered by unit tests.
-   [ ] This PR does not decrease the overall code coverage.
-   [ ] I have reviewed the code comments and documentation for clarity.

### CODE REVIEW

File: pkg/gofr/container/container.go [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// The container package provides a centralized structure to manage common application-level concerns such as
+// logging, connection pools, and service management. This package is designed to facilitate the sharing and
```
Comment (aryanmehrotra) [inline pkg/gofr/container/container.go:2]: can you also include data sources which comprises of different databases, pubsub, filesystem etc.

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]

File: pkg/gofr/container/container.go (hunk 1 above)
Comment (JonasBakys0) [reply pkg/gofr/container/container.go:2]: Sure. Also, I have a question about documentation in go. I saw that `package container` is used in many files which made me wonder what is the best way to document such package? How do I choose which file to add documentation?

Comment (aryanmehrotra) [reply pkg/gofr/container/container.go:2]: `container.go` is the central package which also has the container struct. So it is the best file.

Comment (JonasBakys0) [reply pkg/gofr/container/container.go:2]: I wanted to add some references but couldn't figure out how exactly it works in godoc 😄 
```
Supported data sources:
  - Databases ([cassandra], [clickhouse], [mongo], [solr], [dgraph], [sql])
  - Key value storages [redis] [kafka]
  - Pub/Sub systems ([google], [kafka], [mqtt])
  - File system [file]
```
Results:
image

Even more links to other packages generate false(not found) URL at least locally
Can you help me with this?

Comment (aryanmehrotra) [reply pkg/gofr/container/container.go:2]: You may refer to the following documentation 
https://tip.golang.org/doc/comment#links

**Comment by JonasBakys0** [comment]
I dropped the link idea because it seemed too complicated. For example:
- If I want to use the doc link I face this issue:
image
- And with Links, it's hard to maintain them. What if cassandra package location change
```
// [Cassandra]: https://pkg.go.dev/gofr.dev/pkg/gofr/datasource/cassandra
```

Either way, please correct me if I'm wrong. Go is new to me and don't know how idiomatic documentation should look like in go 😄

**Review by aryanmehrotra — DISMISSED** [review]

File: pkg/gofr/container/container.go [hunk 2]
```diff
@@ -47,6 +59,8 @@ type Container struct {
 	Solr       Solr
 	DGraph     Dgraph
 
+	Dito cassandra
```
Comment (coolwednesday) [inline pkg/gofr/container/container.go:62]: What is this for ?

**Review by coolwednesday — CHANGES_REQUESTED** [review]

**Comment by Umang01-hash** [comment]
@JonasBakys0 the tests and the pipeline is failing due to these errors:
Screenshot 2024-10-03 at 3 07 34 PM


Please fix them.

File: pkg/gofr/container/container.go (hunk 2 above)
Comment (JonasBakys0) [reply pkg/gofr/container/container.go:62]: Good catch, some testing code left.

**Review by Umang01-hash — APPROVED** [review]

**Review by aryanmehrotra — APPROVED** [review]

**Pull request merged**